### ISSUE TITLE
[Tile] Selectable and expandable functionality in one tile

### ISSUE THREAD

**Issue by Kritvi-bhatia17** [carbon-design-system/carbon#16534]
## The problem

- Currently, our tile components are limited in functionality, as they can either be selectable or expandable, but not both.
- We need to determine whether to enhance our tile component to support both selectable and expandable functionalities. 
- **Current assumption:** Combining these features will create a more interactive and user-friendly experience, enabling users to select a tile for actions while also expanding it to view additional information.

### Examples
<img width="617" alt="image" src="https://github.com/carbon-design-system/carbon/assets/158570656/c149621e-e99f-48f8-9829-0aa6506e758f">

---

Product teams mentioning the same request (selectable and expandable functionality in one tile):
1. [Slack thread 1](https://ibm-studios.slack.com/archives/C2K6RFJ1G/p1715540412100209)
2. [Github link](https://github.com/carbon-design-system/carbon/issues/16534#issuecomment-2200220657) 
3. [Slack thread 2](https://ibm-studios.slack.com/archives/C0M053VPT/p1695739692667209)

**Comment by Kritvi-bhatia17** [comment]
We discussed this in our design critique call. Here are the notes from that meeting:

- Further, determine if the product team truly needs a tile with both functionalities.
     - If needed: Consider various use cases and specifications to develop a tile that integrates both selectable and expandable features.
     - If not needed: Maintain the current tile component as it is.
- "There WILL be accessible considerations if you try to put targets inside a selectable tile (target size, label) but my point is that it is ultimately a design problem/question." - MIchael
- Reviewing various components for reference:
     - [Data grid](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-datagrid-clickablerow--clickable-row-story&globals=viewport:basic) (C4IP) 
     - [Dynamic data table](https://react.carbondesignsystem.com/?path=/story/components-datatable-dynamic--playground)
     - [Expandable with interactive tile](https://react.carbondesignsystem.com/?path=/story/components-tile--expandable-with-interactive)

**Comment by Ashley-Johnson28** [comment]
Hello! My team at NS1 Connect would love to be able to have a selectable and expandable tile. Our team would really benefit from this feature enhancement.

**Comment by thyhmdo** [comment]
@Ashley-Johnson28 Can you provide the full-screen visual examples of what you're trying to achieve with selectable + expandable tiles?
